pkg/skaffold/debug/transform.go

```diff
@@ -199,6 +199,51 @@ func transformManifest(obj runtime.Object, retrieveImageConfiguration configurat
 // transformPodSpec attempts to configure a podspec for debugging.
 // Returns true if changed, false otherwise.
 func transformPodSpec(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec, retrieveImageConfiguration configurationRetriever, debugHelpersRegistry string) bool {
+	containers := rewriteContainers(metadata, podSpec, retrieveImageConfiguration, debugHelpersRegistry)
+	timeouts := rewriteProbes(metadata, podSpec) // must rewriteProbes after
+	return containers || timeouts
+}
+
+// rewriteProbes rewrites k8s probes to expand timeouts to 10 minutes to allow debugging local probes.
+func rewriteProbes(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec) bool {
+	annotation, found := metadata.Annotations[DebugConfigAnnotation]
+	if !found {
+		logrus.Warn("no debug config found")
```

Comment:
Debug annotation not present was not surfaced out as a warning before. Is there a reason why we want to show it now?

Reply:
Can we move checking the pod debug annotation at L202 and skip both if no annotation is found?

```
 func transformPodSpec(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec, retrieveImageConfiguration configurationRetriever, debugHelpersRegistry string) bool {
	// skip annotated podspecs — allows users to customize their own image
	if _, found := metadata.Annotations[DebugConfigAnnotation]; found {
		return false
	}
        containers := rewriteContainers(metadata, podSpec, retrieveImageConfiguration, debugHelpersRegistry)
	timeouts := rewriteProbes(metadata, podSpec) // must rewriteProbes after
	return containers || timeouts
}
```

Reply:
You're right: this shouldn't be a warning.  There's an assumption that the probes are only rewritten for containers that were transformed for debugging (I'll make that explicit).